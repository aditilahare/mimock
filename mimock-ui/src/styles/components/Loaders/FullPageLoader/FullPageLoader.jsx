import React from 'react';
import { useRecoilState } from 'recoil';
import { Portal } from 'react-portal';
import SyncLoader from 'react-spinners/SyncLoader';
import showFullPageLoader from 'atoms/showFullPageLoader';
import { LoaderWrapper, LoaderStyle } from './FullPageLoader.style';

function FullPageLoader() {
	const [loading] = useRecoilState(showFullPageLoader);

	return (
		<If condition={loading}>
			<Portal node={document && document.getElementById('app')}>
				<LoaderWrapper data-testid='fullpage-loader'>
					<SyncLoader
						loading={loading}
						color='teal'
						size={30}
						css={LoaderStyle}
					/>
				</LoaderWrapper>
			</Portal>
		</If>
	);
}

export default FullPageLoader;
