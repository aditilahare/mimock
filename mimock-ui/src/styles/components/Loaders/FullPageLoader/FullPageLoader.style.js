import tw from 'tailwind-styled-components';
import { css } from '@emotion/react';

export const LoaderStyle = css`
	display: block;
	margin: 0 auto;
`;

export const LoaderWrapper = tw.div`
    w-screen
    h-screen
    flex
    justify-center
    items-center
    bg-gray-200
    opacity-75
    z-999
    fixed
    top-0
    bottom-0
    left-0
    right-0
`;
