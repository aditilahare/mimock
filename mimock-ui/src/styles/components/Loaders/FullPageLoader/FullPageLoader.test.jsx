import React from 'react';
import { render } from '@testing-library/react';
import { useRecoilState } from 'recoil';
import FullPageLoader from './FullPageLoader';

jest.mock('recoil', () => ({
	atom: jest.fn(),
	useRecoilState: jest.fn(() => [true, jest.fn()]),
}));

describe('FullPageLoader', () => {
	it('should render FullPageLoader', async () => {
		const tree = await render(<FullPageLoader />);

		const { getByTestId } = tree;

		expect(getByTestId('fullpage-loader')).toBeInTheDocument();

		expect(document.body).toMatchSnapshot();
	});

	it('should not render FullPageLoader when loading is false', async () => {
		useRecoilState.mockImplementationOnce(() => [false, jest.fn()]);

		const tree = await render(<FullPageLoader />);

		const { queryByTestId } = tree;

		expect(queryByTestId('fullpage-loader')).not.toBeInTheDocument();

		expect(document.body).toMatchSnapshot();
	});
});
