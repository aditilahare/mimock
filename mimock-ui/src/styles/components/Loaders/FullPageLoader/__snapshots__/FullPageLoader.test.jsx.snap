// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FullPageLoader should not render FullPageLoader when loading is false 1`] = `
<body>
  <div />
</body>
`;

exports[`FullPageLoader should render FullPageLoader 1`] = `
<body>
  <div />
  <div>
    <div
      class="w-screen h-screen flex justify-center items-center bg-gray-200 opacity-75 z-999 fixed top-0 bottom-0 left-0 right-0"
      data-testid="fullpage-loader"
    >
      <span
        class="css-1njh322-LoaderStyle"
      >
        <span
          class="css-88j6kq"
        />
        <span
          class="css-15xvzy0"
        />
        <span
          class="css-1nqujw"
        />
      </span>
    </div>
  </div>
</body>
`;
